tests/test_models.py

```diff
@@ -432,6 +432,18 @@ def test_score_t_with_score_hrt_equality(self) -> None:
 
         assert torch.allclose(scores_t, scores_hrt, atol=1e-06)
 
+    def test_reset_parameters_constructor_call(self):
+        """Tests whether reset_parameters is called in the constructor."""
+        self.model.reset_parameters_ = None
+        try:
+            self.model.__init__(
```

Comment:
why `self.model.__init__()` instead of just `self.model_cls()`?

Reply:
I need it since I have to destroy the `reset_parameters_` method :sweat_smile:

Reply:
ohhhhh yes you bad boy